src/AutoRest.CSharp/LowLevel/Output/RestClientOperationMethods.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using AutoRest.CSharp.Common.Input;
+using AutoRest.CSharp.Common.Output.Models;
+using AutoRest.CSharp.Common.Output.Models.Responses;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Output.Samples.Models;
+
+namespace AutoRest.CSharp.Output.Models
+{
+    internal record RestClientOperationMethods
+    (
+        Method CreateRequest,
+        Method? CreateNextPageMessage,
+        Method? Protocol,
+        Method? ProtocolAsync,
+        Method? Convenience,
+        Method? ConvenienceAsync,
```

Comment:
should we consider some changes in this model to include the case that one protocol method corresponds to multiple convenience methods?
Such as the case described here: https://github.com/Azure/autorest.csharp/issues/3916

Reply:
This separation into individual methods is needed to preserve the different order and different comment logic. With both gone, we can use a simple collection of methods.